docs/shared/recipes/running-tasks/customizing-inputs.md

```diff
@@ -44,136 +48,282 @@ To understand some of the examples below, let's imagine the following simple wor
+2. `^production` is prefixed with `^` and applies the to projects which `myreactapp` depends on: `shared-ui`.
+
+All of the above inputs are taken into consideration when the following command is run:
 
 ```shell
-nx build myreactapp
+nx test myreactapp
 ```
 
-...then because we defined
+## Viewing the Inputs of a Task
 
-- `"inputs": ["default",...]` - it will invalidate the cache of `myreactapp` whenever some src file of `myreactapp` itself changes
-- `"inputs": [..., "^default"]` - it will in addition invalidate the cache of `myreactapp` whenever a file of `shared-ui` (or any of its dependencies) changes
+You can view the configuration for a task of a project by adding the `--graph` flag when running the command:
 
-## Exclude files from invalidating cache
+```shell
+nx build myreactapp --graph
+```
+
+This will show the task graph executed by Nx when running the command.
+
+Clicking the task will open a tooltip which lists out all of the inputs of the task. A button within the tooltip will also reveal more details about the configuration for the project which the task belongs to.
+
+Use this tool to help understand what inputs are being used by Nx to determine when the cache can be used.
+
+{% callout title="Note" type="info" %}
+If no `inputs` are specified at all, Nx will default to looking at all files of a project and its dependencies. This is a rather cautious approach. This might cause Nx to re-run a task in some cases where the cache could have been used instead but it will always give you correct output.
+{% /callout %}
+
+## Configuring Inputs
+
+Inputs of a task are configured in the `inputs` array on the target. This can be done in several places:
 
-Sometimes you might want to exclude specific project files so that they don't invalidate the cache for a given target. For example, we want spec/test files to invalidate the `test` target (by explicitly including `*.spec.ts` files), but we might want to optimize and exclude them from our `build` target. Hence, whenever we just change a test file, our `build` cache would still be working.
+- Nx Plugins often infer inputs for tasks which run other tools.
+- The `inputs` array can be configured in the `targetDefaults` for a set of targets.
+- The `inputs` array can be configured for a specific target in the project configuration file.
+
+{% tabs %}
+{% tab label="Nx Plugins" %}
+Nx plugins specified in the `plugins` array in `nx.json` will define inputs for the tasks which it infers to be cacheable.
 
-Here's how we could define that, starting from our default situation:
+For example, the `@nx/webpack/plugin` plugin, will add the following inputs for tasks which use `webpack` to compile:
 
-```jsonc {% fileName="nx.json" highlightLines=[4, 5, 6, 7, 11] %}
+```jsonc
+  "inputs": [
+    "production", // Production source siles of the project
+    "^production", // Production source files of dependencies
```

Comment:
This might be relevant outside this PR, but if plugins automatically infer using the `production` namedInput, and it doesnt exist in `nx.json` - we will throw an error. 

https://github.com/nrwl/nx/blob/3333ef49684c2f1540ba63408bcc942c99426043/packages/nx/src/native/tasks/inputs.rs#L201-L210